spec/models/theme_spec.rb

```diff
@@ -1347,6 +1347,62 @@ def included_settings(id)
         "deletions" => [{ "key" => "setting_that_will_be_removed", "val" => 1023 }],
       )
     end
+
+    it "does not raise an error and does not create `ThemeSettingsMigration` record for out of sequence migration when `allow_out_of_sequence_migration` kwarg is set to true" do
+      migration_field.update!(value: <<~JS)
+        export default function migrate(settings) {
+          settings.set("integer_setting", 2);
+          return settings;
+        }
+      JS
+
+      second_migration_field =
+        Fabricate(
+          :migration_theme_field,
+          name: "0001-some-other-migration-name",
+          theme: theme,
+          value: <<~JS,
+          export default function migrate(settings) {
+            settings.set("integer_setting", 3);
+            return settings;
+          }
+        JS
+          version: 1,
+        )
+
+      expect do theme.migrate_settings end.to raise_error(Theme::SettingsMigrationError)
+
+      theme.migrate_settings(allow_out_of_sequence_migration: true)
+
+      expect(theme.theme_settings_migrations.count).to eq(1)
+      expect(theme.theme_settings_migrations.first.theme_field_id).to eq(migration_field.id)
+      expect(theme.get_setting("integer_setting")).to eq(3)
+    end
+
+    it "allows custom migration fields to be run by specifing the `fields` kwarg" do
+      expect do theme.migrate_settings(fields: []) end.not_to change {
+        theme.theme_settings_migrations.count
+      }
+
+      migration_field.update!(value: <<~JS)
+        export default function migrate(settings) {
+          settings.set("integer_setting", 2);
+          return settings;
+        }
+      JS
+
+      second_migration_field =
+        Fabricate(:migration_theme_field, theme: theme, value: <<~JS, version: 2)
+          export default function migrate(settings) {
+            settings.set("integer_setting", 3);
+            return settings;
+          }
+        JS
+
+      theme.migrate_settings(fields: [second_migration_field])
+
+      expect(theme.get_setting("integer_setting")).to eq(3)
```

Comment:
Should we use the format to check that we are actually changing from 2 to 3? Or even remove the previous:

```ruby
      migration_field.update!(value: <<~JS)
        export default function migrate(settings) {
          settings.set("integer_setting", 2);
          return settings;
        }
      JS
```

And just check it's changing from nil (or is it 0 by default?) to 3?

Reply:
This assertion is checking that another migration field that is out of sequence is allowed to run. In this case, `second_migration_field` is the migration that is out of sequence and sets the `integer_setting` to `3` so we expect the value to be 3.

Reply:
What Im telling you, is that if I write the spec like this:

```ruby
    it "allows custom migration fields to be run by specifing the `fields` kwarg" do
      expect do theme.migrate_settings(fields: []) end.not_to change {
        theme.theme_settings_migrations.count
      }

      # migration_field.update!(value: <<~JS)
      #   export default function migrate(settings) {
      #     settings.set("integer_setting", 2);
      #     return settings;
      #   }
      # JS

      second_migration_field =
        Fabricate(:migration_theme_field, theme: theme, value: <<~JS, version: 2)
          export default function migrate(settings) {
            settings.set("integer_setting", 3);
            return settings;
          }
        JS

      theme.migrate_settings(fields: [second_migration_field])

      expect(theme.get_setting("integer_setting")).to eq(3)
    end
    ```
    
    It still works

Reply:
O good call 👍 

I added the follow assertions instead.

```
      expect(theme.theme_settings_migrations.count).to eq(1)
      expect(theme.theme_settings_migrations.first.theme_field_id).to eq(second_migration_field.id)
 ```